### PR TITLE
updated spire unit test based on grpc update

### PR DESCRIPTION
**Pull request by pxp928**
# Changes

Address the issue where an update from google.golang.org/grpc from 1.51.0 to 1.53.0 causes the spire tests to fail. See PR: https://github.com/tektoncd/pipeline/pull/6125

Initially, the [spire server](https://github.com/tektoncd/pipeline/blob/main/pkg/spire/controller.go#L102) was not instantiated because it was not needed for the test. Passing in an empty `sc.config.ServerAddr` did not cause an error but with the grpc update, a check was added for the endpoint and returns an error if empty:
https://github.com/grpc/grpc-go/blob/master/internal/resolver/passthrough/passthrough.go#L34-L36

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [x] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [x] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/spire/spire_test.go [hunk 1]
```diff
@@ -145,6 +147,7 @@ func TestCheckHashSimilarities(t *testing.T) {
 
 	cfg := &config.SpireConfig{}
 	cfg.SocketPath = wl.Addr()
+	cfg.ServerAddr = "notImplemented"
```
Comment (pritidesai) [inline pkg/spire/spire_test.go:150]: NIT: suggest declaring a constant for a string value "notImplemented"

**Comment by pritidesai** [comment]
These changes are superset of PR #6125 addressing a broken unit test. We could close the other PR (#6125) once this merged.

/lgtm

**Comment by pxp928** [comment]
@pritidesai @jerop Addressed comment and ready for review

**Review by jerop — COMMENTED** [review]
/lgtm

**Pull request merged**